src/server/editorServices.ts

```diff
@@ -692,17 +692,31 @@ namespace ts.server {
             return this.findExternalProjectByProjectName(projectName) || this.findConfiguredProjectByProjectName(toNormalizedPath(projectName));
         }
 
-        getDefaultProjectForFile(fileName: NormalizedPath, ensureProject: boolean): Project | undefined {
-            let scriptInfo = this.getScriptInfoForNormalizedPath(fileName);
-            if (ensureProject && (!scriptInfo || scriptInfo.isOrphan())) {
-                this.ensureProjectStructuresUptoDate();
-                scriptInfo = this.getScriptInfoForNormalizedPath(fileName);
-                if (!scriptInfo) {
-                    return Errors.ThrowNoProject();
+        getDefaultProjectForFile(fileName: NormalizedPath, ensureProject = false): Project | undefined {
+            const scriptInfo = this.getScriptInfoForNormalizedPath(fileName);

```

Comment:
I would have guessed we'd want all projects for the path.  Is that just out of scope for this PR?

Reply:
Do not change the public method's default here.

Reply:
Tracking with #25058

Reply:
This should return the same result as before if `ensureProject` is provided, though previously it didn't have a default and was non-optional.

Reply:
The parameter ```ensureProject``` was intentional so that we could ensure the passed in parameter and intent (syntactic/semantic/project update needed/not needed)etc